src/Components/Login/Login.jsx

```diff
@@ -0,0 +1,303 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { FormattedMessage } from 'react-intl'
+import LoginInner from './LoginInner.jsx'
+import ConfirmationDialog from '../Base/Dialogs/ConfirmationDialog'
+import KeyHandler from '../Base/KeyHandler'
+import { LOGIN_MNEMONIC, LOGIN_PRIVATE_KEY } from '../Constants'
+
+
+const ethers = require('ethers')
+const Wallet = ethers.Wallet
+
+const keyHandler = new KeyHandler()
+
+const styles = theme => ({
+    wrapper: {
+        display: 'flex',
+        maxWidth: '800px',
+        flexDirection: 'column',
+        alignItems: 'center',
+        alignSelf: 'center',
+        justifyContent: 'center'
+    },
+    logo: {
+        display: 'flex',
+        alignSelf: 'center',
+        padding: '10px 5px',
+        marginBottom: '1em',
+        marginTop: '1em'
+    },
+    logoImage: {
+        maxHeight: '64px'
+    },
+    actions: {
+        display: 'flex',
+        justifyContent: 'flex-end'
+    },
+    button: {
+        flex: '0 auto',
+        margin: theme.spacing.unit,
+    },
+    extendedIcon: {
+      marginLeft: theme.spacing.unit,
+    },
+    loginHeader: {
+        display: 'flex',
+        justifyContent: 'space-between',
+        alignItems: 'center'
+    }
+  })
+
+class Login extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            login: LOGIN_MNEMONIC.toString(),
+            key: '',
+            mnemonic: '',
+            password: '',
+            confirmPassword: '',
+            dialogs: {
+                error: {
+                    open: false,
+                    title: '',
+                    message: ''
+                }
+            }
+        }
+    }
+
+    signUp = () => {
+        let loginType = this.getLoginTypeNumber()
+
+        if (loginType === LOGIN_PRIVATE_KEY)
+            this.signUpPrivateKey()
+        else if (loginType === LOGIN_MNEMONIC)
+            this.signUpMnemonic()
+    }
+
+    getLoginTypeNumber() {
+        return parseInt(this.state.login)
+    }
+
+    signUpPrivateKey = () => {
+        try {
+            const ethWallet = new Wallet(this.state.key)
+            keyHandler.set({
+                vetPubAddress: ethWallet.address,
+                vetPrivateKey: ethWallet.privateKey,
+                address: ethWallet.address,
+                privateKey: ethWallet.privateKey,
+                password: this.state.password
+            })
+            this.props.history.push('/')
+        } catch (e) {
+            let text =
+                "Invalid private key. Please make sure you're entering a valid private key"
+            this.toggleErrorDialog(true, 'Error', text)
+        }
+    }
+
+
+    signUpMnemonic = () => {
+        try {
+            const vetWallet = Wallet.fromMnemonic(this.state.mnemonic, "m/44'/818'/0'/0/0")
+            const wallet = Wallet.fromMnemonic(this.state.mnemonic)
+            console.log('signUpMnemonic', vetWallet, wallet)
```

Comment:
We could clean this up from here as well as in `onPassphraseListener()` in `NewWallet.jsx`